src/web_peer_connection.cpp

```diff
@@ -86,16 +86,20 @@ web_peer_connection::web_peer_connection(peer_connection_args const& pack
 	shared_ptr<torrent> tor = pack.tor.lock();
 	TORRENT_ASSERT(tor);
 
-	// we always prefer downloading 1 MiB chunks
-	// from web seeds, or whole pieces if pieces
-	// are larger than a MiB
-	int preferred_size = 1024 * 1024;
+	// we prefer downloading large chunks from web seeds,
+	// but still want to be able to split requests
+	int preferred_size = m_settings.get_int(settings_pack::urlseed_max_request_bytes);
 
+	int min_size = 1024 * 1024;
 	// if the web server is known not to support keep-alive.
-	// request even larger blocks at a time
-	if (!web.supports_keepalive) preferred_size *= 4;
-
-	prefer_contiguous_blocks((std::max)(preferred_size / tor->block_size(), 1));
+	// request at least 4MiB
+	if (!web.supports_keepalive)
+		min_size *= 4;
+	// we want to have at least piece size to prevent block based requests
+	min_size = (std::max)(tor->torrent_file().piece_length(), min_size);
+	preferred_size = (std::max)(preferred_size, min_size);
```

Comment:
wrapping ``std::min`` in parenthesis is a way to work around windows.h introducing a ``min`` and ``max`` macro, which interferes. It appears that this is not a problem within libtorrent translation units, but should still be used in headers (since those are exposed to client code). We've been transitioning to just use the plain ``std::min(...)`` call in .cpp files. so you don't need it here